File: docs/snippets/all/archetypes/annotation_context_segmentation.py
"""Log a segmentation image with annotations."""

import numpy as np

import dalaran as dl

dl.init("dalaran_example_annotation_context_segmentation", spawn=True)

# Create a simple segmentation image
image = np.zeros((200, 300), dtype=np.uint8)
image[50:100, 50:120] = 1
image[100:180, 130:280] = 2

# Log an annotation context to assign a label and color to each class
dl.log(
    "segmentation",
    dl.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]),
    static=True,
)

dl.log("segmentation/image", dl.SegmentationImage(image))

File: docs/snippets/all/archetypes/annotation_context_segmentation.cpp
// Log a segmentation image with annotations.

#include <dalaran.hpp>

#include <algorithm> // fill_n
#include <vector>

int main(int argc, char* argv[]) {
    const auto rec =
        dalaran::RecordingStream("dalaran_example_annotation_context_segmentation");
    rec.spawn().exit_on_failure();

    // create an annotation context to describe the classes
    rec.log_static(
        "segmentation",
        dalaran::AnnotationContext({
            dalaran::AnnotationInfo(1, "red", dalaran::Rgba32(255, 0, 0)),
            dalaran::AnnotationInfo(2, "green", dalaran::Rgba32(0, 255, 0)),
        })
    );

    // create a segmentation image
    const int HEIGHT = 200;
    const int WIDTH = 300;
    std::vector<uint8_t> data(WIDTH * HEIGHT, 0);
    for (auto y = 50; y < 100; ++y) {
        std::fill_n(data.begin() + y * WIDTH + 50, 70, static_cast<uint8_t>(1));
    }
    for (auto y = 100; y < 180; ++y) {
        std::fill_n(
            data.begin() + y * WIDTH + 130,
            150,
            static_cast<uint8_t>(2)
        );
    }

    rec.log(
        "segmentation/image",
        dalaran::SegmentationImage(data.data(), {WIDTH, HEIGHT})
    );
}
